src/stores/video.ts

```diff
@@ -342,76 +461,130 @@ export const useVideoStore = defineStore('video', () => {
+    const entries = Object.values(videoProcessingDetails.value)
+    const totalProgress = entries.reduce((acc, curr) => acc + curr.progress, 0)
+    return (totalProgress / (totalFilesToProcess.value * 100)) * 100
+  })
+
+  const debouncedUpdateFileProgress = useDebounceFn((filename: string, progress: number, message: string) => {
+    updateFileProgress(filename, progress, message)
+  }, 100)
 
-    const dateStart = new Date(info.dateStart)
-    const dateFinish = new Date(info.dateFinish)
+  const processVideoChunksAndTelemetry = async (hashes: string[]): Promise<void> => {
+    totalFilesToProcess.value = hashes.length
+    videoProcessingDetails.value = {}
 
-    await tempVideoChunksDB.iterate((videoChunk, chunkName) => {
-      if (chunkName.includes(recordingHash)) {
-        chunks.push({ blob: videoChunk as Blob, name: chunkName })
+    const tasks = hashes.map(async (hash) => {
+      const info = unprocessedVideos.value[hash]
+      if (!info) return
+
+      const chunks: { blob: Blob; name: string }[] = []
+      if (info.dateFinish === undefined) {
+        info.dateFinish = info.dateLastRecordingUpdate
       }
-    })
+      debouncedUpdateFileProgress(info.fileName, 1, 'Processing video.')
 
-    // As we advance through the processing, we update the last processing update date, so consumers know this is ongoing
-    updateLastProcessingUpdate(recordingHash)
+      const dateStart = new Date(info.dateStart)
+      const dateFinish = new Date(info.dateFinish)
 
-    if (chunks.length === 0) {
-      throw new Error(`No video chunks found for video ${recordingHash}.`)
-    }
+      debouncedUpdateFileProgress(info.fileName, 30, 'Grouping video chunks.')
+      await tempVideoChunksDB.iterate((videoChunk, chunkName) => {
+        if (chunkName.includes(hash)) {
+          chunks.push({ blob: videoChunk as Blob, name: chunkName })
+        }
+      })
 
-    // Make sure the chunks are sorted in the order they were created, not the order they are stored
-    chunks.sort((a, b) => {
-      const splitA = a.name.split('_')
-      const splitB = b.name.split('_')
-      return Number(splitA[splitA.length - 1]) - Number(splitB[splitB.length - 1])
-    })
+      // As we advance through the processing, we update the last processing update date, so consumers know this is ongoing
+      updateLastProcessingUpdate(hash)
 
-    updateLastProcessingUpdate(recordingHash)
+      if (chunks.length === 0) {
+        throw new Error(`No video chunks found for video ${hash}.`)
+      }
 
-    const chunkBlobs = chunks.map((chunk) => chunk.blob)
+      debouncedUpdateFileProgress(info.fileName, 30, 'Sorting video chunks.')
+      // Make sure the chunks are sorted in the order they were created, not the order they are stored
+      chunks.sort((a, b) => {
+        const splitA = a.name.split('_')
+        const splitB = b.name.split('_')
+        return Number(splitA[splitA.length - 1]) - Number(splitB[splitB.length - 1])
+      })
```

Comment:
This was also already here.

Reply:
As discussed in private, the differ on VSCode is being able to better point the changes, so we will use it to review this PR for now.